### DISCUSSION TITLE
Accessing the app instance with Quasar 2 (Vue 3)

### DISCUSSION THREAD

**Discussion by gfgman** [quasarframework/quasar#11363] [CLI - PWA mode] [answered]
Hello,

I'm migrating to Vue 3, and can't seem to find a way to access the app instance to migrate code like:

```
import Vue from 'vue';
Vue.mixin(blah);
```
Am I supposed to access the app instance only from "boot" folder plugins, or is there a way to access it from any file (in preFetch functions for example)?

Thank you.

**Comment by metalsadman** [comment, marked as answer]
https://quasar.dev/quasar-cli/boot-files#anatomy-of-a-boot-file
```js
export default ({ app, router, store }) => {
  // something to do
  app.mixin(...)
}
```

**Reply by gfgman** [reply]
Ok, so I guess the answer is that I can only do that from boot files... I suppose I could export the app object from a boot file though...

**Reply by salueii** [reply]
I have the same problem. Access to app instance is not possible from JavaScript file other than boot file ??

**Reply by yusufkandemir** [reply]
If you need the app instance outside of a boot file, you probably either didn't understand boot files or are doing something wrong. Please explain in detail what you are trying to do.

**Reply by salueii** [reply]
I set global instance of axios in boot file something like this 
```
const api = axios.create({ baseURL: '' })
app.config.globalProperties.$api = api
```
and want to access $api in other JavaScript file.  I thought by accessing app instance i could access to the $api in JavaScript file.

**Reply by yusufkandemir** [reply]
I don't think you would want to access the `app` elsewhere to do `app.config.globalProperties.$api.get('/some/endpoint')`. That would be annoying to use.

Since this is JavaScript after all, you can just simply export `api` and use it in other places, as shown here:
https://quasar.dev/quasar-cli-vite/ajax-requests#introduction
